tests/languages/docker_test.py

```diff
@@ -1,14 +1,134 @@
+import builtins
+import json
+from typing import List
 from unittest import mock
+from unittest.mock import mock_open
+
+import pytest
 
 from pre_commit.languages import docker
 
 
 def test_docker_fallback_user():
     def invalid_attribute():
         raise AttributeError
+
     with mock.patch.multiple(
-        'os', create=True,
-        getuid=invalid_attribute,
-        getgid=invalid_attribute,
+            'os', create=True,
+            getuid=invalid_attribute,
+            getgid=invalid_attribute,
     ):
         assert docker.get_docker_user() == ()
+
+
+class TestInDocker:
+
+    @pytest.fixture
+    def mock_file_fixture(self):
+        return lambda read_data: \
+            mock.patch.object(
+                builtins, 'open',
+                new_callable=mock_open,
+                read_data=read_data,
+            )
+
+    def test_in_docker_no_file(self, mock_file_fixture):
+        with mock_file_fixture(None) as m:
+            m.side_effect = FileNotFoundError
+
+            assert docker._is_in_docker() is False
+            m.assert_called()
+
+    def test_in_docker_docker_in_file(self, mock_file_fixture):
+        with mock_file_fixture(b'tdockert'):
+            assert docker._is_in_docker() is True
+
+    def test_in_docker_docker_not_in_file(self, mock_file_fixture):
+        with mock_file_fixture(b'testtest'):
+            assert docker._is_in_docker() is False
+
+
+class TestDockerPath:
+    @pytest.fixture
+    def in_docker(self):
+        with mock.patch.object(docker, '_is_in_docker', return_value=True):
+            yield
+
+    @pytest.fixture
+    def not_in_docker(self):
+        with mock.patch.object(docker, '_is_in_docker', return_value=False):
+            yield
+
+    def test_not_in_docker_returns_same(self, not_in_docker):
+        assert docker._get_docker_path('abc') == 'abc'
+
+    def test_in_docker_no_binds_same_path(self, in_docker):
+        binds_list: List[str] = []
+        output_string = json.dumps([{'HostConfig': {'Binds': binds_list}}])
+        with mock.patch.object(
+                docker, 'cmd_output_b',
+                return_value=(0, output_string, b''),
+        ):
+            assert docker._get_docker_path('abc') == 'abc'
+
+    def test_in_docker_binds_path_equal(self, in_docker):
+        binds_list = [
+            '/opt/my_code:/project',
+        ]
+        output_string = json.dumps([{'HostConfig': {'Binds': binds_list}}])
+        with mock.patch.object(
+                docker, 'cmd_output_b',
+                return_value=(0, output_string, b''),
```

Comment:
this `output_string` is of the wrong type -- it should be bytes

Reply:
doned